python/ray/serve/_private/replica.py

```diff
@@ -693,12 +703,45 @@ def __init__(
         self._init_kwargs = init_kwargs
         self._is_function = inspect.isfunction(deployment_def)
         self._deployment_id = deployment_id
-        self._event_loop = event_loop
-        self._delete_lock = asyncio.Lock()
+        self._destructor_called = False
 
         # Will be populated in `initialize_callable`.
         self._callable = None
 
+        # Start the `_user_code_event_loop_thread` singleton if needed.
+        if self._user_code_event_loop_thread is None:
+
+            def _run_user_code_event_loop():
+                asyncio.set_event_loop(self._user_code_event_loop)
```

Comment:
Would suggest to add a comment to elaborate why we need this one

Reply:
good idea done